interfaces/builtin/network_manager.go

```diff
@@ -0,0 +1,444 @@
+         to the agents themselves. -->
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.AgentManager"/>
+
+    <!-- Root-only functions -->
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager"
+          send_member="SetLogging"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager"
+          send_member="Sleep"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager.Settings"
+          send_member="LoadConnections"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager.Settings"
+          send_member="ReloadConnections"/>
+
+    <deny own="org.freedesktop.NetworkManager.dnsmasq"/>
+    <deny send_destination="org.freedesktop.NetworkManager.dnsmasq"/>
+</policy>
+
+<limit name="max_replies_per_connection">1024</limit>
+<limit name="max_match_rules_per_connection">2048</limit>
+`)
+
+type NetworkManagerInterface struct{}
+
+func (iface *NetworkManagerInterface) Name() string {
+	return "network-manager"
+}
+
+func (iface *NetworkManagerInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus, interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *NetworkManagerInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus:
+		return nil, nil
+	case interfaces.SecurityAppArmor:
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := []byte("")
+		if release.OnClassic {
+			// If we're running on classic NetworkManager will be part
+			// of the OS snap and will run unconfined.
+			new = []byte("unconfined")
+		} else {
+			new = slotAppLabelExpr(slot)
+		}
+		snippet := bytes.Replace(networkManagerConnectedPlugAppArmor, old, new, -1)
+		return snippet, nil
+	case interfaces.SecuritySecComp:
+		return networkManagerConnectedPlugSecComp, nil
+	case interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *NetworkManagerInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return networkManagerPermanentSlotAppArmor, nil
+	case interfaces.SecuritySecComp:
+		return networkManagerPermanentSlotSecComp, nil
+	case interfaces.SecurityUDev:
+		return nil, nil
+	case interfaces.SecurityDBus:
+		return networkManagerPermanentSlotDBus, nil
```

Comment:
First of all we will publish the network-manager snap only for Ubuntu Core in the store so that people can't install it on the desktop at all. If then still somebody does its his and not our problem I would say.